packages/grafana-data/src/dataframe/DataFrameJSON.ts

```diff
@@ -0,0 +1,196 @@
+import { DataFrame, FieldType, FieldConfig, Labels, QueryResultMeta } from '../types';
+import { ArrayVector } from '../vector';
+import { guessFieldTypeFromNameAndValue } from './processDataFrame';
+
+/**
+ * The JSON transfer object for DataFrames.  Values are stored in simple JSON
+ *
+ * @alpha
+ */
+export interface DataFrameJSON {
+  /**HACK: this will get removed, but will help transition telegraf streaming
+   *
+   * In telegraf, this will be: ${name}${labels}
+   */
+  key?: string;
+
+  /**
+   * The schema defines the field type and configuration.
+   */
+  schema?: DataFrameSchema;
+
+  /**
+   * The field data
+   */
+  data?: DataFrameData;
+}
+
+/**
+ * @alpha
+ */
+export interface DataFrameData {
+  /**
+   * A columnar store that matches fields defined by schema.
+   */
+  values: any[][];
+
+  /**
+   * Since JSON cannot encode NaN, Inf, -Inf, and undefined, these entities
+   * are decoded after JSON.parse() using this struct
+   */
+  entities?: Array<FieldValueEntityLookup | null>;
```

Comment:
This is what you suggested in slack:
```
"entities": {
   "NaN": [[21, [1, 3]]],
   "Inf": [[21, [4]]],
   "NegInf": [[21, [0]]]
}
```

@leeoniya any thoughts?

Reply:
this structure makes it more expensive to ask "does this series have any entities".

generally i expect `entities` to be absent entirely. if this case is atypical, i prefer the positional uniformity in structure between values, entities, enums, bases, factors. i've suggested maybe using `0` instead of `null` to hold the position, and maybe the same for any values that will get entity-replaced, this turns 4 bytes into 1.

i'm thinking to leave the current encoding as-is and see what real use-cases look like. i assume this isn't gonna be a public API commitment and the wire "compression" format can be swapped later.